File: atCoder/abc187_b/main.m
# tle
1;
function ans=solve(v)
  n = v(1);
  pts = reshape( v(2:end), [2,n]);
  ans = 0;
  for i = 1:n-1
    pi = pts(:,i);
    for j = i+1:n
      pj = pts(:,j);
      if abs(pi(2) - pj(2))>abs(pi(1) - pj(1)) continue end
      ans += 1;
    end
  end
end

v = fscanf(stdin, "%d");
fprintf(stdout, "%d\n", solve(v));

File: atCoder/abc187_b/main.nim
import strscans

var n, x, y: int
discard scanf(stdin.readLine(), "$i", n)
var pts=newSeq[(int,int)](n)
for i in 0 ..< n: 
  discard scanf(stdin.readLine(), "$i $i", x,y)
  pts[i]=(x,y)

var ans = 0
for i in  0 .. n-2:
  var (xi, yi) = pts[i]
  for j in i+1 .. n-1:
    var (xj, yj) = pts[j]
    if abs(yi - yj)>abs(xi - xj): continue
    ans += 1

echo(ans)
